packages/twenty-server/src/modules/invitation/listeners/invitation.listener.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@nestjs/common';
+import { OnEvent } from '@nestjs/event-emitter';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Repository } from 'typeorm';
+
+import { WorkspaceService } from 'src/engine/core-modules/workspace/services/workspace.service';
+import { InvitationWorkspaceEntity } from 'src/modules/invitation/standard-objects/invitation.workspace-entity';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { User } from 'src/engine/core-modules/user/user.entity';
+import { ObjectRecordCreateEvent } from 'src/engine/integrations/event-emitter/types/object-record-create.event';
+
+@Injectable()
+export class InvitationListener {
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    private readonly workspaceRepository: Repository<Workspace>,
+    @InjectRepository(User, 'core')
+    private readonly userRepository: Repository<User>,
+    private readonly workspaceService: WorkspaceService,
+  ) {}
+
+  @OnEvent('invitation.created')
```

Comment:
Related to the long comment below on 'workspace schema', I feel like because you built this in the metadata world you were forced to use an event-based pattern which wouldn't have been your initial inclination if you had built this without the limitations of working in the workspace schema. We do also have a 'pre-hook' / 'post-hook' mechanism available in the context of the workspace schema but it's not something we're proud of and I wouldn't recommend going this route either.